Formula/azure-cli@2.0.rb

```diff
@@ -0,0 +1,54 @@
+require "tmpdir"
+
+class AzureCliAT20 < Formula
+  include Language::Python::Virtualenv
+
+  desc "Microsoft Azure CLI 2.0"
+  homepage "https://docs.microsoft.com/en-us/cli/azure/overview"
+  url "https://azurecliprod.blob.core.windows.net/releases/azure-cli_packaged_0.2.9.tar.gz"
+  sha256 "6f949162d6bbf6848177cc009d60520c12cf2c1dfc068e9e5bbeac0ff6e55e4f"
+  head "https://github.com/Azure/azure-cli.git"
+
+  depends_on :python
+
+  def install
+    virtualenv_create(libexec)
+    bin_dir = libexec/"bin"
+
+    # Install CLI components and dependencies
+    Dir.mktmpdir do |tmp_dir|
+      components = [
+        "src/azure-cli",
+        "src/azure-cli-core",
+        "src/azure-cli-nspkg",
+        "src/azure-cli-command_modules-nspkg",
+      ] + Pathname.glob("src/command_modules/azure-cli-*/")
+
+      # Create wheel distribution of included components
+      components.each do |c|
+        Dir.chdir(c) { system bin_dir/"python", "setup.py", "bdist_wheel", "-d", tmp_dir }
+      end
+
+      # Install CLI from wheel distributions
+      system bin_dir/"pip", "install", "azure-cli", "-f", tmp_dir
+    end
+
+    # Generate executable
+    File.write("#{bin_dir}/az", "#!/usr/bin/env bash\n#{bin_dir}/python -m azure.cli \"$@\"")
```

Comment:
Because this Formula is installed within a Python virtualenv, so we must use the `python` binary matching that virtualenv. The "main" Python doesn't know about the packages installed by this Formula.